src/frequenz/sdk/actor/power_distributing/_batteries_status.py

```diff
@@ -6,33 +6,76 @@
 
 import asyncio
 import logging
+from dataclasses import dataclass
 from typing import Dict, Set
 
-from ..._internal.asyncio import AsyncConstructible
-from ...microgrid._battery import BatteryStatus, BatteryStatusTracker
+from frequenz.channels import Broadcast, Receiver
+from frequenz.channels.util import MergeNamed
+
+from ..._internal.asyncio import cancel_and_await
+from ...microgrid._battery import BatteryStatus, BatteryStatusTracker, RequestResult
 
 _logger = logging.getLogger(__name__)
 
 
-class BatteriesStatus(AsyncConstructible):
-    """Return status of batteries in the pool.
+@dataclass
+class Status:
+    """Status of the batteries."""
 
-    To create an instance of this class you should use `async_new` class method.
-    Standard constructor (__init__) is not supported and using it will raise
-    `NotSyncConstructible` error.
-    """
+    working: Set[int]
+    """Set of working battery ids."""
+
+    uncertain: Set[int]
+    """Set of batteries that should be used only if there are no working batteries."""
+
+    def get_working_batteries(self, batteries: Set[int]) -> Set[int]:
+        """From the given set of batteries return working batteries.
+
+        Args:
+            batteries: Set of batteries
+
+        Returns:
+            Subset with working batteries.
+        """
+        working = self.working & batteries
+        if len(working) > 0:
+            return working
+        return self.uncertain & batteries
 
-    # This is instance attribute.
-    # Don't assign default value, because then it becomes class attribute.
-    _batteries: Dict[int, BatteryStatusTracker]
 
-    @classmethod
-    async def async_new(
-        cls,
+class _BatteryStatusChannel:
+    """Helper class to create battery status channel."""
+
+    def __init__(self, battery_id: int) -> None:
+        """Create class instance.
+
+        Args:
+            battery_id: current battery id
+        """
+        self.battery_id: int = battery_id
+
+        self.name: str = f"battery-{self.battery_id}-status"
+        channel = Broadcast[BatteryStatus](self.name)
+
+        # We need only latest status, so maxsize=1
+        receiver_name = f"{self.name}-receiver"
+        self.receiver = channel.new_receiver(name=receiver_name, maxsize=1)
+
+        self.sender = channel.new_sender()
```

Comment:
I changed it to dataclass, renamed to _BatteryStatusChannelHelper, and I described that the channels has only 1 receiver.

It is just internal, helper class. And it is very simple, so I decided to leave it as class.

Returning tuple doesn't work, because I also need channel name as string. So I would return Tuple with 3 elements and that is a lot.
```
def _create_battery_status_channel(
    battery_id: int,
) -> tuple[Sender[BatteryStatus], Receiver[BatteryStatus]]:
```

Reply:
I'd say it would be more readable if this gets inlined,  but I guess it is just a matter of style.